src/cli/agent.rs

```diff
@@ -48,10 +48,13 @@ pub struct Agent {
     /// The configuration source for a management server.
     #[clap(subcommand)]
     pub provider: Option<crate::config::Providers>,
-    /// If specified, will use the first node address found with the specified
-    /// type
+    /// If specified, filters the available gameserver addresses to the one that
+    /// matches
+    ///
+    /// This takes the form `<address_type>`, and optionally a specifier for the
+    /// IP address kind following a `:` as in `<address_type>:<ipv4|ipv6>`.
```

Comment:
I think we should split this up into two arguments rather than have a syntax, there's nothing stopping a provider from having a `:` in their address type. We can still keep the address selector struct, the only thing I would change is having the address selector be created from the two arguments.

Reply:
Ok